src/sentry/integrations/utils/code_mapping.py

```diff
@@ -349,54 +321,23 @@ def _matches_current_code_mappings(self, src_file: str, frame_filename: FrameFil
             if src_file.startswith(f"{code_mapping.source_path}/")
         )
 
-    def _potential_match_with_transformation(
-        self, src_file: str, frame_filename: FrameFilename
-    ) -> bool:
-        """Determine if the frame filename represents a source code file.
-
-        Languages like Python include the package name at the front of the frame_filename, thus, we need
-        to drop it before we try to match it.
-        """
-        match = False
-        # For instance:
-        #  src_file: "src/sentry/integrations/slack/client.py"
-        #  frame_filename.full_path: "sentry/integrations/slack/client.py"
-        # This should not match:
-        #  src_file: "src/sentry/utils/uwsgi.py"
-        #  frame_filename: "sentry/wsgi.py"
-        split = src_file.split(f"/{frame_filename.file_and_dir_path}")
-        if any(split) and len(split) > 1:
-            # This is important because we only want stack frames to match when they
-            # include the exact package name
-            # e.g. raven/base.py stackframe should not match this source file: apostello/views/base.py
-            match = (
-                split[0].rfind(f"/{frame_filename.root}") > -1 or split[0] == frame_filename.root
-            )
-        return match
-
-    def _potential_match_no_transformation(
-        self, src_file: str, frame_filename: FrameFilename
-    ) -> bool:
-        # src_file: static/app/utils/handleXhrErrorResponse.tsx
-        # full_name: ./app/utils/handleXhrErrorResponse.tsx
-        # file_and_dir_path: app/utils/handleXhrErrorResponse.tsx
-        return src_file.rfind(frame_filename.file_and_dir_path) > -1
-
     def _potential_match(self, src_file: str, frame_filename: FrameFilename) -> bool:
         """Tries to see if the stacktrace without the root matches the file from the
         source code. Use existing code mappings to exclude some source files
         """
         # Exit early because we should not be processing source files for existing code maps
-        if self._matches_current_code_mappings(src_file, frame_filename):
+        if self._matches_current_code_mappings(src_file):
             return False
 
-        match = False
-        if frame_filename.full_path.endswith(".py"):
-            match = self._potential_match_with_transformation(src_file, frame_filename)
+        src_file_items = src_file.split("/")
+        frame_items = frame_filename.full_path.split("/")
+        if len(src_file_items) > len(frame_items):
+            return src_file.rfind(frame_filename.full_path) > -1
         else:
-            match = self._potential_match_no_transformation(src_file, frame_filename)
-
-        return match
+            num_relevant_items = max(1, len(src_file_items) - 1)
+            relevant_items = frame_items[(len(frame_items) - num_relevant_items) :]
+            relevant_path = "/".join(relevant_items)
+            return src_file.rfind(relevant_path) > -1
```

Comment:
This is new-ish logic to determine if there is a potential match between a github repo source path and a stack trace. If the source file is longer (by # of directories, not character-length), then simply check if the stack trace path is a trailing substring. Otherwise, truncate the stack trace's file path to be at most N - 1 items longer, where N is the number of items in the github repo source path. This is to provide support for languages that use absolute paths in their stack traces, rather than relative path.